lib/modules/manager/pep621/processors/pdm.ts

```diff
@@ -39,4 +48,75 @@ export class PdmProcessor implements PyProjectProcessor {
 
     return deps;
   }
+
+  async updateArtifacts(
+    updateArtifact: UpdateArtifact
+  ): Promise<UpdateArtifactsResult[] | null> {
+    const { config, updatedDeps, packageFileName } = updateArtifact;
+
+    const isLockFileMaintenance = config.updateType === 'lockFileMaintenance';
+
+    // abort if no lockfile is defined
+    const lockFileName = getSiblingFileName(packageFileName, 'pdm.lock');
+    try {
+      const existingLockFileContent = await readLocalFile(lockFileName, 'utf8');
+      if (is.nullOrUndefined(existingLockFileContent)) {
+        logger.debug('No pdm.lock found');
+        return null;
+      }
+
+      const toolConstraint: ToolConstraint = {
+        toolName: 'pdm',
+        constraint: config.constraints?.pdm,
+      };
+
+      const execOptions: ExecOptions = {
+        docker: {},
+        toolConstraints: [toolConstraint],
+      };
+
+      // on lockFileMaintenance do not specify any packages and update the complete lock file
+      // else only update specific packages
+      let packageList = '';
+      if (!isLockFileMaintenance) {
+        packageList = updatedDeps
+          .map((value) => value.packageName ?? value.depName)
+          .join(' ');
+      }
+      const cmd = `pdm update ${packageList}`;
+      await exec([cmd], execOptions);
```

Comment:
```suggestion
      await exec(cmd, execOptions);
```
you can pass a string as single command